### PR TITLE
filter balance transfers and staking join candidates

### PR DESCRIPTION
**Pull request by asiniscalchi**
## **Type**
enhancement, tests

___

## **Description**
- Implemented a new system configuration for the runtime, including a custom `BaseCallFilter` to restrict specific operations like token transfers and joining as a collator.
- Added a new `system` module configuration.
- Removed the old `frame_system::Config` implementation from `lib.rs` and adjusted related imports and traits.
- Enhanced test infrastructure by making `ALICE` and `BOB` constants public for broader use in tests.
- Added tests to validate the restrictions imposed by `BaseCallFilter`, ensuring operations like token transfers and joining as a collator are blocked.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>mod.rs</strong><dd><code>Import System Module in Configurations</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

ownership-chain/runtime/laos/src/configs/mod.rs
- Added `system` module import to the configurations module.

</details>
    

  </td>
  <td><a href="https://github.com/freeverseio/laos/pull/418/files#diff-8d3c73b179fe7958404091b4442985abc69ab49c211e78ac1e5d6a6b48aa3d7e">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>system.rs</strong><dd><code>Implement System Config and Operation Restrictions</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

ownership-chain/runtime/laos/src/configs/system.rs
<li>Implemented <code>frame_system::Config</code> for the <code>Runtime</code>.<br> <li> Defined <code>BaseCallFilter</code> to restrict certain operations like token <br>transfers and joining as a collator.<br> <li> Added comprehensive tests to ensure the restrictions are enforced.<br>

</details>
    

  </td>
  <td><a href="https://github.com/freeverseio/laos/pull/418/files#diff-54c6ed1dcb381ca93e7e38ea7d16d047e3258f18281f39c155b380b81d888e46">+216/-0</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>lib.rs</strong><dd><code>Adjust Runtime Configuration for New System Implementation</code></dd></summary>
<hr>

ownership-chain/runtime/laos/src/lib.rs
<li>Removed the previous <code>frame_system::Config</code> implementation.<br> <li> Adjusted imports and traits due to the new system configuration.<br>

</details>
    

  </td>
  <td><a href="https://github.com/freeverseio/laos/pull/418/files#diff-e71ce79c6361a064296c98f1244126573a4ae185adc40f1afca2fc27e5e74c38">+4/-52</a>&nbsp; &nbsp; </td>
</tr>                    
</table></td></tr><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>mod.rs</strong><dd><code>Update Test Constants Visibility</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

ownership-chain/runtime/laos/src/tests/mod.rs
- Made `ALICE` and `BOB` constants public for testing purposes.

</details>
    

  </td>
  <td><a href="https://github.com/freeverseio/laos/pull/418/files#diff-973242a0b8b4602b6287a2170770f896132d769da0452b4b597dd3451108decd">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: ownership-chain/runtime/laos/src/configs/system.rs [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+	type SS58Prefix = SS58Prefix;
+	/// The action to take on a Runtime Upgrade
+	type OnSetCode = cumulus_pallet_parachain_system::ParachainSetCode<Self>;
+	type MaxConsumers = frame_support::traits::ConstU32<16>;
+}
+
+pub struct BaseCallFilter;
+impl Contains<RuntimeCall> for BaseCallFilter {
+	fn contains(c: &RuntimeCall) -> bool {
+		use pallet_balances::Call::*;
```
Comment (magecnion) [inline ownership-chain/runtime/laos/src/configs/system.rs:59]: usually we place imports at the beginning of the file

File: ownership-chain/runtime/laos/src/configs/system.rs [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+				transfer_allow_death { .. } => false,
+				_ => true,
+			},
+			RuntimeCall::Vesting(inner_call) => match inner_call {
+				// Vested transfes are not allowed.
+				vested_transfer { .. } => false,
+				_ => true,
+			},
+			RuntimeCall::ParachainStaking(inner_call) => match inner_call {
+				// Staking is not allowed.
```
Comment (dastanfv) [inline ownership-chain/runtime/laos/src/configs/system.rs:78]:
```suggestion
				// New candidates are not allowed.
```

File: ownership-chain/runtime/laos/src/configs/system.rs (hunk 1 above)
Comment (dastanfv) [reply ownership-chain/runtime/laos/src/configs/system.rs:59]: it's ok to put this here if these imports are only used in this context

**Review by magecnion — APPROVED** [review]

**Pull request merged**